### ISSUE TITLE
allow logging to a different syslog facility

### ISSUE THREAD

**Issue by anarcat** [fail2ban/fail2ban#1980]
### Environment:

- Fail2Ban version (including any possible distribution suffixes): 0.9.6-2
- OS, including release name/version: Debian GNU/Linux stretch 9.1 x86_64
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:

It would be nice to be able to change the syslog facility used by fail2ban. Right now, it is hardcoded to `DAEMON` in the source code, which is a fine default for most daemons. But considering that fail2ban often (always?) deals with authentication systems, I feel that `AUTH` is a more appropriate value here. Indeed, here is the documentation of facilities from [syslog(3)](https://manpages.debian.org/syslog.3)

       LOG_AUTH       security/authorization messages

       LOG_AUTHPRIV   security/authorization messages (private)

       LOG_CRON       clock daemon (cron and at)

       LOG_DAEMON     system daemons without separate facility value

It would seem more relevant to use `LOG_AUTH`. Not wanting to create a huge bikeshed discussion about such a cosmetic change, I figured it was better to just make that customizable. Some users, for example, may *only* use fail2ban for mail purposes so they may prefer to see the logs end up in `LOG_MAIL`. Or maybe they want to use a fail2ban-specific facility to filter logs on the server, using `LOG_LOCAL0`.

Having that flexibility would be very useful in some configurations.

#### Steps to reproduce

I was looking for entries in /var/log/auth.log, found them in /var/log/daemon.log instead, which makes it hard to correlate with `sshd` logs.

#### Expected behavior

I would expect to see a `logfacility = DAEMON` line in the configuration, it could look something like this:

```
# Option: logfacility
# Notes.: Sets the syslog "facility". 
# syslog facilities allow the syslog daemon to sort output to different logfiles or servers.
# Allowed values are defined by the Python syslog handler, define here:
# https://docs.python.org/3/library/logging.handlers.html#logging.handlers.SysLogHandler.encodePriority
# Values: [ AUTH | DAEMON | LOCAL0 | ... ]  Default: DAEMON
#
logfacility = DAEMON
```

#### Observed behavior

No such configuration exists, obviously.

#### Any customizations done to /etc/fail2ban/ configuration

This is obviously relevant only with the following configuration:

```
logtarget = SYSLOG
```

Thanks!

**Comment by sebres** [comment]
👍 for the possible PR
👎 for new option (`logfacility = DAEMON`), because I would prefer to use already available `logtarget` with something like this:
```
logtarget = SYSLOG[facility=DAEMON]
```
This can be easy extracted using function `extractOptions`, and provide opportunity for enhancement of other targets using similar way.

**Comment by anarcat** [comment]
that seems totally clunky to me, but okay. not sure what other targets would support such parameters... the "file" parameter is already just a string, and i suspect such a change would make configs ambiguous...

**Comment by sebres** [comment]
> that seems totally clunky to me
> i suspect such a change would make configs ambiguous

This is the way how different things will be configured by fail2ban currently: filters, actions, backends etc.

> not sure what other targets would support such parameters... 

If you'd posted this issue a few days earlier, I'd made #1970 differently:
```INI
logtarget = stdout[datetime=off]
```
So the current target `sysout` (e. g. used to log into systemd-journal) would be just an alias for `stdout[datetime=off]`.

But also other options would be possible, for example define own format:
```INI
logtarget = stdout[log-format="%%(asctime)-15s | %%(thread) | %%(levelname)-5.5s | %%(message)s"]
```

> the "file" parameter is already just a string...

Let the file be a file, at least at the moment, but tomorrow it may get more:
```INI
logtarget = /var/log/fail2ban.log[owner="root:admin", permissions="u=rw,g=r,o="]
```

**Comment by sebres** [comment]
Implemented in #1989 for >= 0.10, I'll merge it in the near future.
